### PR TITLE
Revert "borg buckle nerf"

### PR DESCRIPTION
**Pull request by carshalash**
Actions merge got merged so, this is redundant.

### CODE REVIEW

**Comment by actioninja** [comment]
lol

**Comment by TheVekter** [comment]
I mean, you're not wrong. 👍

**Comment by 81Denton** [comment]
#47851 got merged before #47866, also this is pretty funny.

**Comment by ATH1909** [comment]
> #47851 got merged before #47866, also this is pretty funny.

I'd prefer keeping the former PR over keeping the latter PR, but if we HAVE to keep the latter PR, we might as well undo the former PR.

**Comment by PKPenguin321** [comment]
It being instant was a dumb oversight and it stunning was never meant to be used for combat the way it was

**Comment by carshalash** [comment]
> > #47851 got merged before #47866, also this is pretty funny.
> 
> I'd prefer keeping the former PR over keeping the latter PR, but if we HAVE to keep the latter PR, we might as well undo the former PR.

I feel the same way but, ya know.

**Comment by ninjanomnom** [comment]
Personally it makes more sense that you can't just instantly buckle people to yourself as a borg. If you're doing this out of concern for combat balance then I'd argue buckling shouldn't be viable in the middle of combat. If there are other reasons this revert is wanted then I'd be interested in what those reasons are.

**Comment by ATH1909** [comment]
> Personally it makes more sense that you can't just instantly buckle people to yourself as a borg. If you're doing this out of concern for combat balance then I'd argue buckling shouldn't be viable in the middle of combat. If there are other reasons this revert is wanted then I'd be interested in what those reasons are.

Well, because spinstunning only knocks people down now (unless you're an emagged borg who's flinging someone against a wall or someone else), your victims can now just crawl to the side/away before you can re-buckle them to yourself. Since they're not likely to have their weapon(s) out while riding you and flashes are melee weapons anyway, there's not much point to knocking someone down as a cyborg without chaining it somehow to make them drop the flash that they're gonna whip out immediately after you spinknockdown them.

**Comment by wesoda25** [comment]
What about this being an unwanted combat feature is so hard to understand

**Comment by ninjanomnom** [comment]
>...there's not much point to knocking someone down as a cyborg without chaining it...

I agree with the arguments that it shouldn't be a thing especially viable in combat at all. Saying it has not much point in combat is a point in favor for keeping the delay.

I asked if there are any non combat reasons why this pr should be merged so I can consider those.

**Comment by carshalash** [comment]
> > ...there's not much point to knocking someone down as a cyborg without chaining it...
> 
> I agree with the arguments that it shouldn't be a thing especially viable in combat at all. Saying it has not much point in combat is a point in favor for keeping the delay.
> 
> I asked if there are any non combat reasons why this pr should be merged so I can consider those.

Rescuing people that are in the middle of being injured, you know.. Law 1 and all? The reason why I was mostly behind xzabers version of the nerf.

**Comment by PKPenguin321** [comment]
A 0.5 second delay doesn't stop you from quickly moving injured people. It's only stopping you from snagging moving people, and generally if somebody is moving away from you they don't want to be buckled.

**Comment by carshalash** [comment]
> A 0.5 second delay doesn't stop you from quickly moving injured people. It's only stopping you from snagging moving people, and generally if somebody is moving away from you they don't want to be buckled.

🤔 you know bad boys can just drag people they're beating to death so you cant buckle them right?

**Comment by ExcessiveUseOfCobblestone** [comment]
that's the point of the slowdown drag nerf...

**Comment by ninjanomnom** [comment]
Similar to combat, being able to buckle someone who is actively being dragged away from you seems bad but I could go either way.

As PKPenguin says, when rescuing someone this cooldown shouldn't make a difference except when some force is acting against you in which case there being time to buckle someone doesn't make sense to me nor would I want this ability to be used in such a situation.

Honestly I'd be in favor of making all buckles take half a second.

**Comment by PKPenguin321** [comment]
>🤔 you know bad boys can just drag people they're beating to death so you cant buckle them right?

Okay, then if they're moving away from you either they don't want you bucking them or whoever is moving them doesn't want you buckling them, same thing

**Comment by carshalash** [comment]
> that's the point of the slowdown drag nerf...

The slowdown is more helpful to stop the person doing the harm, it doesn't really help too much for actually freeing said victim as the slowdown will apply to who ever is dragging. It's pretty easy to work around for people to get around through physical force to make them fuck off, shooting, batoning, toolboxing or general shove spamming.

**Comment by Shadowflame909** [comment]
Shadowflame909 thinks all nerfs must be fair whether your sillicon or carbon.

This evens the playing field. Sillicons can zoom away with an instant fireman carry (Dare I say the original, and also the original mount!) Yet humans cannot?

Revert the fireman carry then revert this

**Comment by ATH1909** [comment]
> It being instant was a dumb oversight and it stunning was never meant to be used for combat the way it was

Just because borg spinning might not have been _intended_ to be combat-relevant doesn't mean that it _shouldn't_ be combat-relevant. Emergent gameplay, my dude.

It just seems a bit... _wrong_ that anyone with sunglasses, insulated gloves, and a flash can almost completely wall opposing non-emagged cyborgs now (yes, it's not technically a complete walling, since you can drag in securitrons, slip them once with your extinguisher (before needing to refill your extinguisher again), etc., but it's very close to being one). If the buckling delay was removed, you could probably try some cool-looking tricks where you spinknockdown someone, then spinknockdown them across to the other side of the room immediately after they pull out their flash, then do some more fancy stuff from there, but nope, they can just crawl away now.

To put it another way, merging TWO borgspin nerfs was definitely overkill, and it's really hurt the overall combat viability of non-emagged cyborgs in a manner that lowers the non-theoretical skill ceiling for them, IMO.

**Comment by ATH1909** [comment]
> Shadowflame909 thinks all nerfs must be fair whether your sillicon or carbon.
> 
> This evens the playing field. Sillicons can zoom away with an instant fireman carry (Dare I say the original, and also the original mount!) Yet humans cannot?
> 
> Revert the fireman carry then revert this

Damnit, Shadowflame909, you're not helping our case here.

**Comment by CatDerpington** [comment]
> Shadowflame909 thinks all nerfs must be fair whether your sillicon or carbon.
> 
> This evens the playing field. Sillicons can zoom away with an instant fireman carry (Dare I say the original, and also the original mount!) Yet humans cannot?
> 
> Revert the fireman carry then revert this

Agree. In my brief time on the server I’ve never seen either spin or buckle used appropriately. If anything I’ve seen borgs abusing both. Buckling people for no reason and racing off into restricted areas. Pulling people away from security. Stun locking people with spin.

Borgs have a lot of things they can do to subdue people. Bolting doors etc. I don’t see how these two things aren’t balanced as is in the code.

If anything it seems to me this argument is largely about adding functions back that are and will largely be used for griefing.

**Comment by CatDerpington** [comment]
> > It being instant was a dumb oversight and it stunning was never meant to be used for combat the way it was
> 
> Just because borg spinning might not have been _intended_ to be combat-relevant doesn't mean that it _shouldn't_ be combat-relevant. Emergent gameplay, my dude.
> 
> It just seems a bit... _wrong_ that anyone with sunglasses, insulated gloves, and a flash can almost completely wall opposing non-emagged cyborgs now (yes, it's not technically a complete walling, since you can drag in securitrons, slip them once with your extinguisher (before needing to refill your extinguisher again), etc., but it's very close to being one). If the buckling delay was removed, you could probably try some cool-looking tricks where you spinknockdown someone, then spinknockdown them across to the other side of the room immediately after they pull out their flash, then do some more fancy stuff from there, but nope, they can just crawl away now.
> 
> To put it another way, merging TWO borgspin nerfs (that weren't designed with each other in mind) was DEFINITELY overkill, and gutting the viability of borgspins has really hurt the overall combat viability of non-emagged cyborgs in a manner that has lowered the (non-theoretical) skill ceiling for them, IMO.

It’s my understanding borgs should not be involved in combat period. Could be wrong.

**Comment by carshalash** [comment]
You're meant to interfere with sec if you see them committing human harm, and if you don't take people into areas when they demand you to you'll be violating law 2 unless you're taking them to the armory.  The stunning was already addressed in actionninjass pr to make it a knockdown instead of a stun. You could probably use this as well.
https://tgstation13.org/wiki/AI_modules

**Comment by CatDerpington** [comment]
> You're meant to interfere with sec if you see them committing human harm, and if you don't take people into areas when they demand you to you'll be violating law 2 unless you're taking them to the armory. The stunning was already addressed in actionninjass pr to make it a knockdown instead of a stun. You could probably use this as well.
> https://tgstation13.org/wiki/AI_modules

That’s fine but human harm is up to interpretation. I’ve seen borgs grab someone who was taken to brig by them, stunned by sec in the process of being arrested and ran away with the person. If anything Borg speed should be nerfed as well as uploading these changes. They mostly are used for grief. 

Further can’t you still stun lock them? Knockdown roll up. Buckle. Spin. Knockdown. Rinse repeat.

**Comment by carshalash** [comment]
Buckling isnt a stun, you're able to interact with items on you or on the ground near you. It's the same as being slipped, you're also able to unbuckle with that nice resist button you can bind to anything.

**Comment by CatDerpington** [comment]
> Buckling isnt a stun, you're able to interact with items on you or on the ground near you. It's the same as being slipped, you're also able to unbuckle with that nice resist button you can bind to anything.

I’m aware. Buckling. Spin. Buckle. Spin. Kind of a stun-lock if you’re spamming those.

**Comment by carshalash** [comment]
If you don't understand the important difference of knockdown and stuns, it's on you not me.

**Comment by CatDerpington** [comment]
> If you don't understand the important difference of knockdown and stuns, it's on you not me.

You’re claiming you can’t continuously knockdown someone using spin? Isn’t it 3 seconds vs 5 seconds? 

I’ll admit maybe I’m not grasping it but buckling. Spin. Using Borg speed to cover the distance buckle and spin seems entirely possible.

**Comment by carshalash** [comment]
It's the difference between being slipped and being stunned.

**Comment by SkeletalElite** [comment]
> > If you don't understand the important difference of knockdown and stuns, it's on you not me.
> 
> You’re claiming you can’t continuously knockdown someone using spin? Isn’t it 3 seconds vs 5 seconds?
> 
> I’ll admit maybe I’m not grasping it but buckling. Spin. Using Borg speed to cover the distance buckle and spin seems entirely possible.

You can't infinitely spin someone with just knockdown. They can still move and use items while knocked down.

**Comment by ninjanomnom** [comment]
This is currently fulfilling the intended purpose and I've seen no arguments made that bring up additional points I don't already know about or care about. If you think borgs need more tools are their disposal to deal with flash proof people then make another pr with your suggested change but this one at least isn't appropriate.

**Comment by FancyCyclops** [comment]
:thinking:

**Comment by carshalash** [comment]
Thank you for not reading any od my arguments ninja, very cool.

**Comment by CatDerpington** [comment]
They did. They addressed them. Working as intended. Deal with it.

**Comment by ExcessiveUseOfCobblestone** [comment]
> 
> 
> > that's the point of the slowdown drag nerf...
> 
> The slowdown is more helpful to stop the person doing the harm, it doesn't really help too much for actually freeing said victim as the slowdown will apply to who ever is dragging. It's pretty easy to work around for people to get around through physical force to make them fuck off, shooting, batoning, toolboxing or general shove spamming.

Being gimped by the default laws so that you can't deter someone from harming in an effective way like performing "self"defense on behalf of someone else shouldn't really make/break a feature though.

On the code side we need to assume they AREN'T bound by asimov since it's a supported usecase, regardless of how often lawchanges occur ingame/on your server.
